src/ItemsContext.js

```diff
@@ -4,21 +4,35 @@ import firebase from 'firebase';
 export const ItemsContext = createContext();
 
 const ItemsProvider = props => {
-  //set useState([]) for list
   const [items, setItems] = useState([]);
   const db = firebase.firestore();
+  const userToken = localStorage.getItem('token');
 
   useEffect(() => {
     db.collection('items')
+      .where('token', '==', userToken)
       .get()
       .then(snapshot => {
         const list = snapshot.docs.map(doc => doc.data());
         setItems(list);
       });
-  }, []);
+  }, [db, userToken]);
 
   return (
-    <ItemsContext.Provider value={items}>
+    <ItemsContext.Provider
+      value={{
```

Comment:
We had a similar idea. Great minds think alike!! :wink:

Reply:
Yes! I forgot about `onSnapshot` and all that it can do. I think when we decide to merge into main, we should continue using `onSnapshot` instead of the function Fabrice and I made. Makes the code cleaner. 🌟